### PR TITLE
vitorvezani/fwi-280/fail-fast-on-git-fetching-process

### PR DESCRIPTION
**Pull request by vitorvezani**
## Description
fail fast on git info fetching and file write

## Checklist

* [X] I have updated the CHANGELOG of any plugins updated
* [X] I have updated the version file of any plugins updated
* [X] I have added tests wherever appropriate
* [X] I have labeled this PR with a label per plugin updated

### CODE REVIEW

File: plugins/ci/pkg/ci/ci.go [hunk 1]
```diff
@@ -267,28 +268,24 @@ func SendResults(reports []models.ReportInfo, resources []models.Resource, confi
...
+			logrus.Fatalf("Unable to write contents for %s: %v", file.field, err)
 		}
 	}
 	w.Close()
 
 	repoDetails, err := getGitInfo(configurationObject.Options.RepositoryName, configurationObject.Options.BaseBranch)
 	if err != nil {
-		logrus.Warn("Unable to get git details")
-		return results, err
+		logrus.Fatalf("Unable to get git details: %v", err)
```
Comment (rbren) [inline plugins/ci/pkg/ci/ci.go:288]: Can we make this
```
Unable to get git details. Please be sure to run the insights-ci script inside of a valid git repository, with the branch you are scanning checked out. Details: %v
```

**Review by rbren — APPROVED** [review]

File: plugins/ci/pkg/ci/ci.go [hunk 2]
```diff
@@ -342,22 +339,27 @@ func SendResults(reports []models.ReportInfo, resources []models.Resource, confi
 func getGitInfo(repoName, baseBranch string) (gitInfo, error) {
 	info := gitInfo{}
 
+	_, err := GetResultsFromCommand("git", "rev-parse", "--is-inside-work-tree")
+	if err != nil {
+		logrus.Error("Unable to execute git command. Are you in a git repository?")
```
Comment (vitorvezani) [inline plugins/ci/pkg/ci/ci.go:344]: @rbren I would rather do:

`return nil, fmt.Errorf("%v: %v", "Please be sure to run the insights-ci script inside of a valid git repository, with the branch you are scanning checked out.", err)`

here, what do you think? `getGitInfo` may fail for different reasons...

Comment (rbren) [reply plugins/ci/pkg/ci/ci.go:344]: Hmm yeah agreed!

**Comment by rbren** [comment]
It would be good to also add a `Warn` here: https://github.com/FairwindsOps/insights-plugins/blob/06a4992752beb7784fc8bd4db2756831bc2c8dc3/plugins/ci/pkg/ci/ci.go#L49

That way we can log the actual command/args that failed

File: plugins/ci/pkg/ci/ci.go (hunk 2 above)
Comment (vitorvezani) [reply plugins/ci/pkg/ci/ci.go:344]:
```INFO[0000] Updating dependencies for staging            
INFO[0000] Templating: staging                          
INFO[0000] Copying yaml file                            
INFO[0000] Copying yaml file                            
INFO[0000] Audit with Polaris                           
ERRO[0002] Unable to execute command: git rev-parse --is-inside-work-tree 
FATA[0002] Unable to get git details: Please be sure to run the insights-ci script inside of a valid git repository, with the branch you are scanning checked out: exit status 128 ```

**Review by rbren — APPROVED** [review]

**Comment by vitorvezani** [comment]
> It would be good to also add a `Warn` here:
> 
> https://github.com/FairwindsOps/insights-plugins/blob/06a4992752beb7784fc8bd4db2756831bc2c8dc3/plugins/ci/pkg/ci/ci.go#L49
> 
> That way we can log the actual command/args that failed

maybe you missed but I added it there https://github.com/FairwindsOps/insights-plugins/pull/463/files#diff-044cde4902d1869165353b9fc0e73b91be6154df96d8e5ecdbd18fd7a3ed4326R48

**Pull request merged**